space-framework/docs/space-framework.md

```diff
@@ -0,0 +1,92 @@
+The SM uses a library of space provider adaptors to communicate with the space providers. Currently the SF includes 3 space provider adaptors KCP, KIND, and KubeFlex.
+
+### Space provider adaptors
+The space manager uses a set of space provider adaptors that interacts with the space provider. All space provider adaptors implement a simple "space client interface" that includes basic pSpace life cycle operatios. In addition the interface also defines the events that sent from the adaptors to the space manager.
+
+### Space aware client (SAC)
+The Space aware client (msclient) allows clients/controllers to easily get access to the underlying space by simply using the space name. There is no direct interaction between the msclient and the Space  provider, and therefore the msclient is transparent to the specific provider of the spaces. 
+![Space Aware Client](SAC.drawio.svg "Space Aware Client")
+The only requirement is that the space can be accessed through regular Kube APIs when using the appropriate kubeconfig information. 
+
+**Main features**  
+- Constantly watch for changes in the available spaces. The SAC is using an informer on Space objecs using the SF Space client-go APIs.
+    - Holds a cache of the pSpace access info for each Space 
+- Exposes utility functions to get RestConfig by simpy specifying the space name  
+
+Currently the SAC exposes a simple function that supply the config access info for a space. In the future we will add a SAC space aware ClientSets
+
+## Architecture details 
+When a new SpaceProviderDesc is created the SM creates a new namespace that will be used to host all the Space objects representing pSpaces on that space provider.  
+The name of the generated namespace will be `paceprovider-<provider name>`
+
+![Space, SpaceProviderDesc and the Space provider namespace](SF-NS.drawio.svg "SF Namespaces")
+
+Note: In the future the spaces will be created in their own namespace and not in namespace per provider. 
+
+The Space also holds a reference to the SpaceProviderDesc associated with this Space and some capabilities (e.g., the space type). The status of the space holds a secret reference to the secret holding the access information for that pSpace. As described before for managed spaces the SF retrive that information from teh space provider, create the needed secrets and update the Space object with the secret reference
+
+**Discovery**  
+As mentioned before, the Space manager also supports discovery of spaces created out of band (i.e., not through creating Space objects).
+
+Each provider adaptor constantly watch for pSpace creation/deletion events and feeds the SM with those events. In case a pSpace that matches the discovery filtering is detected and there is no corrresponding Space object the SM creates a Space object representing this new detected pSpace and sets its mode to “unmanaged” .
```

Comment:
Presumably the SM _deletes_ Space objects that it discovered too. The omission of this from the text leaves the reader wondering. Also, are there any _update_ cases?

Reply:
Added